WalletWasabi.Fluent/ViewModels/SearchBar/Settings/Setting.cs

```diff
@@ -0,0 +1,68 @@
+using System.Diagnostics.CodeAnalysis;
+using System.Linq.Expressions;
+using System.Reactive;
+using System.Reactive.Linq;
+using System.Windows.Input;
+using ReactiveUI;
+using WalletWasabi.Fluent.Helpers;
+
+namespace WalletWasabi.Fluent.ViewModels.SearchBar.Settings;
+
+public class Setting<TTarget, TProperty> : ReactiveObject
+{
+	private TProperty? _value;
+
+	public Setting([DisallowNull] TTarget target, Expression<Func<TTarget, TProperty>> selector,
+		bool requiresRestart = false)
+	{
+		if (target == null)
+		{
+			throw new ArgumentNullException(nameof(target));
+		}
+
+		if (selector == null)
+		{
+			throw new ArgumentNullException(nameof(selector));
+		}
+
+		if (PropertyHelper<TTarget>.GetProperty(selector) is not { } pr)
```

Comment:
Is this the same as: `if (PropertyHelper<TTarget>.GetProperty(selector) is null)`?

Reply:
Hi @yahiheb! It's equivalent to this:

```
var propertyInfo = PropertyHelper<TTarget>.GetProperty(selector);
if (propertyInfo != null)
{
}
```